### PR TITLE
Evaluate per-study normalization

### PR DESCRIPTION
**Pull request by ben-heil**
Test whether setting each study's mean to zero and standard deviation to one performs better than PLIER's built-in row normalization, which normalizes a given gene across all samples.

Changes:
- Moved `compare_models.ipynb` to the analyses folder
- Created the script `normalize_expression_by_study.py` to do the study-wise normalization
- Modified `run_plier.R` to give the option to not row normalize
- Made the notebook `normalization_analysis.ipynb` to analyze the results
- Added tests for `normalize_expression_by_study.py`

### CODE REVIEW

File: analyses/normalization_analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+       "<ggplot: (8738774078991)>"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ggplot(result_df, aes(x='factor(lv_count)', y='val_acc', color='factor(lv_count)')) +\\\n",
```
Comment (jjc2718) [inline analyses/normalization_analysis.ipynb:261]: May be good to add a title or more informative x-axis label here, it wasn't immediately obvious to me what you were plotting (looks like it's number of PLIER latent variables vs. validation accuracy?)

File: analyses/normalization_analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
...hC8AAAAAMADhCwAAAAAMQPgCAAAAAAMQvgAAAADAAIQvAAAAADAA4QsAAAAADED4AgAAAAADEL4AAAAAwACELwAAAAAwAOELAAAAAAxA+AIAAAAAA/h4uoBrlZqaqlmzZunbb7+Vv7+/unTpokceeSTPer/88ouWLl2q33//XZJUo0YN9e7dW5UqVTK6ZAAAAAAoeWe+5syZI5vNpvnz52vkyJFavHixfvjhhzzrpaWlqUWLFnr33Xe1YMECRUdHa/z48R6oGAAAAABKWPjKzMzUF198occff1wBAQGKiYlRs2bN9Omnn+ZZ9+6771ajRo0UGBgoX19ftWvXTidOnFBKSooHKgcAAABwvStR4evkyZOSpOjoaOeyqlWr6ujRo1e97/79+xUaGqrg4OBiqw8AAAAALqdEXfOVmZkpf39/l/qB3hMAACAASURBVGWBgYHKyMi44v1OnTqlOXPmqF+/fvneHh8fr/j4eOffZrNZERERRS+4ECwWi8v/Swuz4+//WyyymDxTw8XtnJiYKEkKDQ2VJPn4+Kh8+fKeKUyls929ZZ+8td295flxN2/YL9rcWBzfr6y0trs37Nel39/Cw8MVHh7usXqAgihR4atMmTJ5glZaWlqeQHax+Ph4jRo1Sp06ddIDDzyQ7zqrVq3S3LlznX/HxsZqwIAB7im6kErbGTpbVrakBJUNClKo1dcjNcyZM8elnS/Wp0+fy4ZzI5W2dpc8v0/e3u6efn6Kiyf3izY3Fsf3gilt7X6BN73XvaWtgSsxORwOh6eLKKjMzEx1795d06dPV1RUlCRp3rx5SkpK0uDBg/Osn5CQoOHDh6tly5bq1KnTZbfrbWe+goODlZKSIpvN5pEaikOSQ3r5zwRNrFRe5Tz0y6jNZvPqX0ZLW7t7yz55a7t7y/Pjbt6wX7S5sTi+X1lpbXdv2K+L213izBdKhhJ35qthw4ZavHixBg4cqLi4OG3btk1DhgzJs+6F4NWkSZMrBi8p75s1Pj7e4wdIm83m8Rrcyf53xLfbbLJ56MP5Sgdlb2hzqfS1u+T5ffL2dvf081NcPLlftLmxOL4XTGlr9wu89b0OeKsSNeCGJOfp5NjYWI0dO1Y9evRQnTp1JEldunTRTz/9JEn65JNP9Ndff2nNmjXq0qWL878zZ854rHYAAAAA168SdeZLkoKCgjRs2LB8b1u+
```
Comment (jjc2718) [inline analyses/normalization_analysis.ipynb:391]: Do you have any intuition for why the outlier detection models seem to have so much more variance in accuracy/AUC over different validation sets? Maybe class imbalance has more of an effect on them than the other models?

File: brdnet/run_plier.R [hunk 3]
```diff
@@ -33,6 +33,9 @@ parser <- add_argument(parser, 'plierDisease',
 parser <- add_argument(parser, 'outdir', help='The directory to print results to', type='character')
 parser <- add_argument(parser, '-k', help='The number of PLIER PCs to use', 
 					   type='numeric')
+parser <- add_argument(parser, '--noNorm',
+					   help='If this flag is passed, do not normalize the gene expression data',
```
Comment (jjc2718) [inline brdnet/run_plier.R]: I think the spacing is a bit off here (unless this is just a GitHub rendering issue)

File: brdnet/normalize_expression_by_study.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    -------
+    None
+    '''
+
+    healthy_out_path = healthy_original_file_path.rstrip('.tsv') + '_normalized.tsv'
+    disease_out_path = disease_original_file_path.rstrip('.tsv') + '_normalized.tsv'
+
+    healthy_df = normalized_df[healthy_columns]
+    # I don't know of an easier way to select all columns not in healthy_columns
+    disease_df = normalized_df[normalized_df.columns.difference(healthy_columns)]
```
Comment (jjc2718) [inline brdnet/normalize_expression_by_study.py:36]: This seems good to me personally, I think it's explicit/obvious what you're doing which is almost always preferable to slightly faster but less explicit (unless this is a performance bottleneck, but I doubt it is)

**Review by jjc2718 — APPROVED** [review]
Nice work! Just a few small things.

File: analyses/normalization_analysis.ipynb (hunk 1 above)
Comment (ben-heil) [reply analyses/normalization_analysis.ipynb:261]: Good point, added new labels

File: analyses/normalization_analysis.ipynb (hunk 2 above)
Comment (ben-heil) [reply analyses/normalization_analysis.ipynb:391]: That is a good question, I haven't been able to figure it out so far

File: brdnet/run_plier.R (hunk 3 above)
Comment (ben-heil) [reply brdnet/run_plier.R]: Somehow some tabs crept into my code and the interpreter didn't complain because R. It's a github rendering issue, but it also means that something is wrong in my .vimrc somewhere.
